### PR TITLE
Added information about the new date handling in the comparison constraints and Range

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes (https://github.com/symfony/symfony/pull/11673) |
| Applies to | 2.6+ |
| Fixed tickets | - |

This PR finishes #4143

### CODE REVIEW

File: reference/constraints/GreaterThan.rst [hunk 1]
```diff
@@ -24,11 +23,20 @@ than another value, see :doc:`/reference/constraints/LessThan`.
...
-If you want to ensure that the ``age`` of a ``Person`` class is greater
-than ``18``, you could do the following:
+If you want to ensure that the ``age`` of a ``Person`` class is greater than
+``18``, you could do the following:
 
 .. configuration-block::
 
+    .. code-block:: yaml
+
+        # src/SocialBundle/Resources/config/validation.yml
```
Comment (xabbuh) [inline reference/constraints/GreaterThan.rst]: `Acme` is missing

File: reference/constraints/GreaterThan.rst [hunk 2]
```diff
@@ -90,6 +89,191 @@ than ``18``, you could do the following:
             }
         }
 
+Comparing Dates
+---------------
+
+.. versionadded:: 2.6
+    The feature to compare dates was added in Symfony 2.6.
```
Comment (xabbuh) [inline reference/constraints/GreaterThan.rst]: was introduced

**Comment by xabbuh** [comment]
Looks like the ordering of the different formats needs to be fixed (we switched to show annotations, YAML, XML, PHP).

File: reference/constraints/GreaterThan.rst [hunk 3]
```diff
@@ -14,7 +14,6 @@ than another value, see :doc:`/reference/constraints/LessThan`.
 +----------------+---------------------------------------------------------------------------+
 | Options        | - `value`_                                                                |
 |                | - `message`_                                                              |
-|                | - `payload`_                                                              |
```
Comment (xabbuh) [inline reference/constraints/GreaterThan.rst]: The `payload` option should be kept.

File: reference/constraints/GreaterThanOrEqual.rst [hunk 4]
```diff
@@ -23,11 +22,20 @@ in the options. To force that a value is greater than another value, see
...
-If you want to ensure that the ``age`` of a ``Person`` class is greater
-than or equal to ``18``, you could do the following:
+If you want to ensure that the ``age`` of a ``Person`` class is greater than
+or equal to ``18``, you could do the following:
 
 .. configuration-block::
 
+    .. code-block:: yaml
+
+        # src/SocialBundle/Resources/config/validation.yml
```
Comment (xabbuh) [inline reference/constraints/GreaterThanOrEqual.rst]: missing `Acme`

File: reference/constraints/GreaterThanOrEqual.rst [hunk 5]
```diff
@@ -89,6 +88,191 @@ than or equal to ``18``, you could do the following:
             }
         }
 
+Comparing Dates
+---------------
+
+.. versionadded:: 2.6
+    The feature to compare dates was added in Symfony 2.6.
```
Comment (xabbuh) [inline reference/constraints/GreaterThanOrEqual.rst]: was introduced

File: reference/constraints/LessThan.rst [hunk 6]
```diff
@@ -90,6 +89,190 @@ If you want to ensure that the ``age`` of a ``Person`` class is less than
             }
         }
 
+Comparing Dates
+---------------
+
+.. versionadded:: 2.6
+    The feature to compare dates was added in Symfony 2.6.
```
Comment (xabbuh) [inline reference/constraints/LessThan.rst]: was introduced

File: reference/constraints/LessThanOrEqual.rst [hunk 7]
```diff
@@ -23,11 +22,20 @@ the options. To force that a value is less than another value, see
...
-If you want to ensure that the ``age`` of a ``Person`` class is less than
-or equal to ``80``, you could do the following:
+If you want to ensure that the ``age`` of a ``Person`` class is less than or
+equal to ``80``, you could do the following:
 
 .. configuration-block::
 
+    .. code-block:: yaml
+
+        # src/SocialBundle/Resources/config/validation.yml
```
Comment (xabbuh) [inline reference/constraints/LessThanOrEqual.rst]: missing `Acme`

**Comment by javiereguiluz** [comment]
@xabbuh I've just fixed all the issues that you reported and I've also changed he order of the configuration block sot show annotations first.

**Comment by xabbuh** [comment]
@javiereguiluz Seems like the `payload` option would still be removed in the `LessThanOrEqual`, `LessThan` and `GreaterThanOrEqual` constraints.

**Comment by xabbuh** [comment]
Oh, and the code format ordering is wrong the `Range` constraint.

**Comment by javiereguiluz** [comment]
@xabbuh sorry for these issues! Everything should be fixed now.

**Comment by javiereguiluz** [comment]
I think we can now consider this issue as `finished`. Thanks.

**Comment by WouterJ** [comment]
Just a thought (I'm happy to apply it after merging), what about showing one example (the first one) and then doing something like?

```rst
Be aware that PHP will use the server's configured timezone to interpret these
dates. If you want to fix the timezone, append it to the date string (e.g. ``today UTC``).
You can also use relative dates or times. For example, you can check that the
above delivery date starts at least five hours after the current time using ``+5 hours``
instead of ``today``.
```

This will make the section quite a bit shorter, which is a good thing imo (it's a nice feature, but not so important to take almost 200 lines).

**Comment by weaverryan** [comment]
This is awesome! Thanks Javier - merged into the 2.6 branch.

**Comment by WouterJ** [comment]
@weaverryan can you please also comment on my proposal? :smiley_cat:

**Comment by weaverryan** [comment]
@WouterJ missed that somehow! Yes, I'd prefer your change due to length

**Comment by javiereguiluz** [comment]
I'd prefer to maintain the current examples. The may look long but in reality this length is due to showing four different configuration formats. The examples themselves are quite small (except XML of course; XML is always awfully verbose).

**Pull request closed without merging**